### PR TITLE
HPCC-29132 Improve the tagging scripts to support new major versions

### PR DESCRIPTION
**Pull request by ghalliday**
## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by ghalliday** [comment]
@richardkchapman please can you review this.  Before I run it I will test and not push the branches/tags, and check all is correct.

**Comment by ghalliday** [comment]
@richardkchapman Fixed a couple of issues and used to create the new 9.0.0rc1.

File: tools/tagging/gominor.sh [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+   git submodule update --recursive --init --force
+done
+
+echo Press any key to create a new minor version for "$all"
+read -n 1 -s
+
+for f in $all ; do
+   cd $gitroot/$f
+   echo "Process $f"
+   $hpccdir/cmake_modules/go_rc.sh $*
```
Comment (richardkchapman) [inline tools/tagging/gominor.sh:43]: Is this different from gorc.sh ? Does gominor require a parameter? Should gorc.sh pass $* ?

**Review by richardkchapman — APPROVED** [review]
I compared the three (gorc.sh, gominor.sh, and gomajor.sh) and noted a few inconsistencies e.g. gomajor/minor pass $* to go_rc.sh, but gorc does not, gorc.sh has an echo that is missing form the others.

It looks like (as far as the go_rc base script is concerned) you signal a new minor by starting from the master branch, and a new major by starting from the master branch AND passing in a flag. Feels like passing in a --minor flag (and a sanity check that the start branch is appropriate) might be better. At that point you could also common up the three goxxx.sh scripts and decide what to check out first based on the flag. 

Whether any of the above are worth worrying about though is up to you. I will approve on the basis that this PR fine as it is, if you prefer.

**Comment by ghalliday** [comment]
I will make the suggested changes in a follow on PR.

**Pull request merged**